dotCMS/src/main/java/com/dotcms/rest/api/v1/categories/CategoriesResource.java

```diff
@@ -271,6 +271,87 @@ public final CategoryView save(@Context final HttpServletRequest  httpRequest,
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requestAndResponse(httpRequest, httpResponse).rejectWhenNoUser(true).init();
+        final User user = initData.getUser();
+        final Host host = this.categoryHelper.getHost(categoryEditDTO.getSiteId(),
+                () -> this.hostWebAPI.getCurrentHostNoThrow(httpRequest));
+        final PageMode pageMode = PageMode.get(httpRequest);
+
+        Logger.debug(this,
+                () -> "Saving category sortOrder. Request payload is : " + getObjectToJsonString(
+                        categoryEditDTO));
+
+        DotPreconditions.checkArgument(UtilMethods.isSet(categoryEditDTO.getCategoryData()),
+                "The body must send a collection of category inode and sortOrder");
+
+        Category parentCategory = null;
+
+        if (UtilMethods.isSet(categoryEditDTO.getParentInode())) {
+            parentCategory = this.categoryAPI.find(categoryEditDTO.getParentInode(), user,
+                    pageMode.respectAnonPerms);
+        }
+
+        Iterator iterator = categoryEditDTO.getCategoryData().entrySet().iterator();
+
+        while(iterator.hasNext()) {
+            Map.Entry entry = (Map.Entry) iterator.next();
+            String key = (String) entry.getKey();
+            Integer value = (Integer) entry.getValue();
+
+                final Category category = this.categoryAPI.find(key, user, pageMode.respectAnonPerms);
+
+                if (null == category) {
+                    Logger.error(this,
+                            "Category with Id: " + key + " does not exist");
+                } else {
+
+                    category.setSortOrder(value);
+
+                    Logger.debug(this,
+                            () -> "Saving category entity : " + getObjectToJsonString(category));
+                    this.categoryAPI.save(parentCategory, category, user,
+                            pageMode.respectAnonPerms);
+                    Logger.debug(this,
+                            () -> "Saved category entity : " + getObjectToJsonString(category));
+
+                    ActivityLogger.logInfo(this.getClass(), "Saved Category",
+                            "User " + user.getPrimaryKey()
+                                    + "Category: " + category.getCategoryName(),
+                            host.getTitle() != null ? host.getTitle() : "default");
+                }
+            }
+
+        if(parentCategory == null) {
+            return this.paginationUtil.getPage(httpRequest, user, categoryEditDTO.getFilter(),
+                    categoryEditDTO.getPage(), categoryEditDTO.getPerPage());
+        }
+        else{
+           return this.getChildren(httpRequest,httpResponse,categoryEditDTO.getFilter(), categoryEditDTO.getPage(),
+                    categoryEditDTO.getPerPage(),"", categoryEditDTO.getDirection(),categoryEditDTO.getParentInode());
+        }
+    }
```

Comment:
```
this could be:
return parentCategory == null 
                    ? this.paginationUtil.getPage(httpRequest, user, categoryEditDTO.getFilter(),
                    categoryEditDTO.getPage(), categoryEditDTO.getPerPage());
                    : this.getChildren(httpRequest,httpResponse,categoryEditDTO.getFilter(), categoryEditDTO.getPage(),
                    categoryEditDTO.getPerPage(),"", categoryEditDTO.getDirection(),categoryEditDTO.getParentInode());
```

Reply:
yeah